vsphere/resource_vsphere_resource_pool.go

```diff
@@ -0,0 +1,255 @@
+	dc, _ := datacenterFromID(client, "hashi-dc")
+	rp, err := resourcepool.FromPathOrDefault(client, d.Id(), dc)
+	if err != nil {
+		return nil, err
+	}
+	d.SetId(rp.Reference().Value)
+	rpProps, err := resourcepool.Properties(rp)
+	if err != nil {
+		return nil, err
+	}
+	err = flattenResourcePoolConfigSpec(d, &rpProps.Config)
+	if err != nil {
+		return nil, err
+	}
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceVSphereResourcePoolCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning create", resourceVSphereResourcePoolIDString(d))
+	client := meta.(*VSphereClient).vimClient
+	rrp, err := resourcepool.FromID(client, d.Get("root_resource_pool_id").(string))
+	if err != nil {
+		return err
+	}
+	rpSpec := expandResourcePoolConfigSpec(d)
+	rp, err := resourcepool.Create(rrp, d.Get("name").(string), rpSpec)
+	if err != nil {
+		return err
+	}
+	d.SetId(rp.Reference().Value)
+	log.Printf("[DEBUG] %s: Create finished successfully", resourceVSphereResourcePoolIDString(d))
+	return nil
+}
+
+func resourceVSphereResourcePoolRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning read", resourceVSphereResourcePoolIDString(d))
+	client := meta.(*VSphereClient).vimClient
+	rp, err := resourcepool.FromID(client, d.Id())
+	if err != nil {
+		return err
+	}
+	_ = d.Set("name", rp.Name())
+	rpProps, err := resourcepool.Properties(rp)
+	if err != nil {
+		return err
+	}
+	err = flattenResourcePoolConfigSpec(d, &rpProps.Config)
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] %s: Read finishes successfully", resourceVSphereResourcePoolIDString(d))
+	return nil
+}
+
+func resourceVSphereResourcePoolUpdate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning update", resourceVSphereResourcePoolIDString(d))
+	client := meta.(*VSphereClient).vimClient
+	rp, err := resourcepool.FromID(client, d.Id())
+	if err != nil {
+		return err
+	}
+	rpSpec := expandResourcePoolConfigSpec(d)
+	err = resourcepool.Update(rp, d.Get("name").(string), rpSpec)
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] %s: Update finished successfully", resourceVSphereResourcePoolIDString(d))
+	return nil
+}
+
+func resourceVSphereResourcePoolDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning delete", resourceVSphereResourcePoolIDString(d))
+	client := meta.(*VSphereClient).vimClient
+	rp, err := resourcepool.FromID(client, d.Id())
+	if err != nil {
+		return err
+	}
+	err = resourcepool.Delete(rp)
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] %s: Deleted successfully", resourceVSphereResourcePoolIDString(d))
+	return nil
+}
+
+// resourceVSphereResourcePoolIDString prints a friendly string for the
+// vsphere_virtual_machine resource.
+func resourceVSphereResourcePoolIDString(d structure.ResourceIDStringer) string {
+	return structure.ResourceIDString(d, "vsphere_resource_pool")
```

Comment:
Replace bare string here with the defined `resourceVSphereResourcePoolName` const.